System/Process.hs

```diff
@@ -870,6 +869,22 @@ It will therefore behave more portably between operating systems than 'system'.
 The return codes and possible failures are the same as for 'system'.
 -}
 rawSystem :: String -> [String] -> IO ExitCode
-rawSystem cmd args = do
-  (_,_,_,p) <- createProcess_ "rawSystem" (proc cmd args) { delegate_ctlc = True }
+rawSystem cmd args = executeAndWait "rawSystem" (proc cmd args) { delegate_ctlc = True }
+
+-- ---------------------------------------------------------------------------
+-- executeAndWait
+
+-- | Create a new process and wait for it's termination.
```

Comment:
Right, so the basic problem is this:
Say you have two programs `A` and `B`.  Now `A` calls `B`. There are two ways to do this.

1) You can use the normal `CreateProcess` API, which is what normal Windows code do. Using this approach, the current `waitForProcess` works absolutely fine.
2) You can call the emulated POSIX function `_exec`, which of course is supposed to allow the child process to replace the parent. 

With approach `2)` `waitForProcess` falls apart because the Win32's process model does not allow this the same way as linux. `_exec` is emulated by first making a call to `CreateProcess` to spawn `B` and then immediately exiting from `A`. So you have two different processes.

`waitForProcess` is waiting on the termination of `A`. Because `A` is immediately killed, `waitForProcess` will return even though `B` is still running. This is why for instance the testsuite on Windows had lots of file locked errors.

This approach creates a new `Job` and assigned `A` to the job, but also all future processes spawned by `A`. This allows us to listen in on events, such as, when all processes in the job are finished, but also allows us to propagate exit codes from `_exec` calls.

The only reason we need this at all is because we don't interact with just actual native code on Windows, and instead have a lot of ported POSIX code.

I could add a new constructor to `ProcessHandle` and make the decision on what to do internally in the functions. This would probably generate warnings in user code that deconstruct `ProcessHandle` though.